api/lib/application/pre-response-utils.js

```diff
@@ -14,9 +16,17 @@ function handleDomainAndHttpErrors(request, h) {
     return errorManager.handle(request, h, new UnauthorizedError(undefined, 401));
   }
 
+  if (_isError5XX(response) && config.logging.isError5XXLoggingEnabled) {
+    logger.error(response);
```

Comment:
Il ya a aussi la solution d'utiliser le logger Hapi (good-console), utilisé dans 99% des cas (notamment tracer les requêtes et les logs `ops` avec mémoire et CPU), voir le [Wiki](https://1024pix.atlassian.net/wiki/spaces/~879933039/pages/2675310597/Tech+time#Track-4%3A-Pour-ceux-qui-n%E2%80%99ont-pas-froid-aux-yeux)

Les raisons d'utiliser bunyan ne sont pas claires, si quelqu'un en a, je fais un ADR
```suggestion
    request.server.log('error', response);
```

Reply:
Ah merci, je (re?)decouvre `request.server.log`. C'est peut être mieux dans la couche http effectivement.

Reply:
Sur pourquoi Bunyan, je dirais que quand on n'a pas de contexte de requête (i.e. partout sauf dans la couche http) on ne peut pas appeler `request.server.log` ? Si c'est bien le cas on est obligé d'utiliser autre chose.

Reply:
Discussion hors PR : ça me fait penser que dans un projet, quand il y a un module qui s'appelle "logger" j'aime bien l'utiliser. Ça pourrait être ce module qui redirige vers Hapi quand un contexte de requête est dispo, ou on pourrait ajouter des méthodes http aware dans ce module (logger.http.error(request, ...)), pour regrouper et rendre visible les options ? En général pour me rafraîchir les idées je fais un "grep logger", j'aurais donc plus de chance de me rappeler cette autre option.
Aussi, ça nous permettrait de pouvoir vérifier dans un test auto que le logger est appelé dans la couche http (ce qu'on fait ici). Je ne sais pas si mocker `request.server.log` serait ok, il es injecté dans la méthode ce qui est bien, mais son interface ne nous appartient pas, don't mock what you don't own semble s'appliquer ?

Reply:
Y'a plein de moyen de faire, et l'un d'eux est éligible après-midi Tech.
Sur le principe, je pense aussi que le logging devrait être centralisé

Reply:
Aussi, je n'ai trouvé aucune occurrence de request.server dans le projet api, et quinze appels au logger dans la couche http, j'en déduis que l'habitude actuelle c'est plutôt d'appeler logger même dans la couche http.
Si j'ai bien compris, les access logs générés par Hapi utilisent bien le logger Hapi, mais les logs fait par nous explicitement utilisent notre module logger.
Suite à cette discussion, ça me parait bien comme ça ici du coup.

Reply:
En fait, je pense que logger a été utilisé dans les scripts, là ou il n'y a pas le choix (puisque pas de Hapi). Ensuite, si personne ne connaissait server.log, et bien la méïose a fait son travail :wink:

Reply:
ADR créé pour valider l'usage du logger
https://github.com/1024pix/pix/pull/3056